### PR TITLE
Add options for actions left and done for on screen XP

### PR DESCRIPTION
**Pull request by deathbeam**
Add configurable option for on screen XP display mode, what now
includes:
- Actions done
- Actions left
- Xp gained
- Xp left

Signed-off-by: Tomas Slusny <slusnucky@gmail.com>

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpTrackerConfig.java [hunk 1]
```diff
@@ -77,12 +95,12 @@ default int pauseSkillAfter()
 
 	@ConfigItem(
 		position = 4,
-		keyName = "onScreenXpLeft",
-		name = "Display XP Left on-screen",
+		keyName = "onScreenDisplayMode",
+		name = "On-screen tracker display mode",
 		description = "Display remaining experience instead of experience gained on on-screen trackers"
```
Comment (devLotto) [inline runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpTrackerConfig.java]: You forgot the description 😛

Comment (deathbeam) [reply runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpTrackerConfig.java]: Oh oops, updated

File: runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpTrackerConfig.java [hunk 2]
```diff
@@ -77,12 +95,12 @@ default int pauseSkillAfter()
...
-		description = "Display remaining experience instead of experience gained on on-screen trackers"
+		keyName = "onScreenDisplayMode",
+		name = "On-screen tracker display mode",
+		description = "Configures how first line in on screen XP overlay is displayed"
 	)
-	default boolean displayXpLeftOnScreen()
+	default OnScreenDisplayMode onScreenDisplayMode()
 	{
-		return false;
+		return OnScreenDisplayMode.XP_LEFT;
```
Comment (devLotto) [inline runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpTrackerConfig.java]: Default should be XP_GAINED

Comment (deathbeam) [reply runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpTrackerConfig.java]: Updated

File: runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpInfoBoxOverlay.java [hunk 3]
```diff
@@ -96,11 +96,34 @@ public Dimension render(Graphics2D graphics)
...
+				break;
+		}
+
 		final LineComponent xpLine = LineComponent.builder()
-				.left(config.displayXpLeftOnScreen() ? "XP Left:" : "XP Gained:")
-				.right(StackFormatter.quantityToRSDecimalStack(config.displayXpLeftOnScreen()
-					? snapshot.getXpRemainingToGoal() : snapshot.getXpGainedInSession()))
-				.build();
+			.left(leftStr + ":")
+			.right(StackFormatter.quantityToRSDecimalStack(rightNum))
```
Comment (devLotto) [inline runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpInfoBoxOverlay.java]: I also just noticed that I forgot to make it use the more accurate stack representation, maybe you can change it to do that here and for the XP/Hour below. ``StackFormatter.quantityToRSDecimalStack(..., true)``

File: runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpTrackerConfig.java [hunk 4]
```diff
@@ -77,12 +95,12 @@ default int pauseSkillAfter()
 
 	@ConfigItem(
 		position = 4,
-		keyName = "onScreenXpLeft",
-		name = "Display XP Left on-screen",
-		description = "Display remaining experience instead of experience gained on on-screen trackers"
+		keyName = "onScreenDisplayMode",
+		name = "On-screen tracker display mode",
+		description = "Configures how first line in on screen XP overlay is displayed"
```
Comment (devLotto) [inline runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpTrackerConfig.java]: Maybe ``Configures the information displayed in the first line of on-screen XP overlays`` is better 😃

File: runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpTrackerConfig.java [hunk 5]
```diff
@@ -24,13 +24,31 @@
...
+import lombok.AllArgsConstructor;
 import net.runelite.client.config.Config;
 import net.runelite.client.config.ConfigGroup;
 import net.runelite.client.config.ConfigItem;
 
 @ConfigGroup("xpTracker")
 public interface XpTrackerConfig extends Config
 {
+	@AllArgsConstructor
+	public enum OnScreenDisplayMode
```
Comment (devLotto) [inline runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpTrackerConfig.java]: My IntelliJ says ``public`` is redundant here

**Comment by deathbeam** [comment]
Updated

**Pull request merged**